cvetools/cvesearch.go

```diff
@@ -204,16 +204,25 @@ func (cv *CveTools) ScanAppPackage(req *share.ScanAppRequest, namespace string)
 
 // ScanImage helps the Image scanning
 func (cv *CveTools) ScanImage(ctx context.Context, req *share.ScanImageRequest, imgPath string) (*share.ScanResult, error) {
-	var err error
 	result := &share.ScanResult{
-		Provider:        share.ScanProvider_Neuvector,
-		Version:         cv.CveDBVersion,
-		CVEDBCreateTime: cv.CveDBCreateTime,
-		Error:           share.ScanErrorCode_ScanErrNone,
-		Registry:        req.Registry,
-		Repository:      req.Repository,
-		Tag:             req.Tag,
-		Layers:          make([]*share.ScanLayerResult, 0),
+		Provider:           share.ScanProvider_Neuvector,
+		Version:            cv.CveDBVersion,
+		CVEDBCreateTime:    cv.CveDBCreateTime,
+		Error:              share.ScanErrorCode_ScanErrNone,
+		Registry:           req.Registry,
+		Repository:         req.Repository,
+		Tag:                req.Tag,
+		Layers:             make([]*share.ScanLayerResult, 0),
+		ScanTypesRequested: req.ScanTypesRequested,
+	}
+
+	log.WithFields(log.Fields{"scanTypesRequested": req.ScanTypesRequested}).Info("Scan types requested")
+	if req.ScanTypesRequested == nil {
+		log.Info("No scan types requested, defaulting to full scan")
+		req.ScanTypesRequested = &share.ScanTypeMap{
+			Vulnerability: true,
+			Signature:     true,
```

Comment:
For NV controller prior-5.2.2 + this scanner, req.ScanTypesRequested is always nil when ScanImage() is called & then req.ScanTypesRequested.Signature is always set to true
It means line 297
result.SignatureInfo, _, _ = getSatisfiedSignatureVerifiersForImage(rc, req, info, ctx)
is always executed
Is it possible to tell whether the grpc requester is NV controller version 5.2.0/5.2.1 or prior-5.2.0 so that 
scanner does not always set req.ScanTypesRequested.Signature to true when grpc requester is version prior-5.2.0?